modules/Bio/EnsEMBL/Utils/Tree/Interval/Mutable/Node.pm

```diff
@@ -0,0 +1,664 @@
+  
+  return $self->{height};
+}
+
+=head2 left
+
+  Arg []      : none
+  Description : Return the node's left child
+  Returntype  : Bio::EnsEMBL::Utils::Tree::Interval::Mutable::Node
+  Exceptions  : none
+  Caller      : general
+
+=cut
+
+sub left {
+  my $self = shift;
+  $self->{left} = shift if( @_ );
+  
+  return $self->{left};
+}
+
+=head2 right
+
+  Arg []      : none
+  Description : Return the node's right child
+  Returntype  : Bio::EnsEMBL::Utils::Tree::Interval::Mutable::Node
+  Exceptions  : none
+  Caller      : general
+
+=cut
+
+sub right {
+  my $self = shift;
+  $self->{right} = shift if( @_ );
+  
+  return $self->{right};
+}
+
+=head2 search
+
+  Arg [1]     : Bio::EnsEMBL::Utils::Interval
+                The interval to search for overlaps in the tree
+  Description : Search the node and its successors for overlapping
+                intervals with the query
+  Returntype  : Arrayref of Bio::EnsEMBL::Utils::Interval
+  Exceptions  : none
+  Caller      : general
+
+=cut
+
+sub search {
+  my ($self, $i) = @_;
+
+  # if interval is to the right of the rightmost point of any interval in this node and
+  # all its children, there won't be any matches
+  return [] if $i->start > $self->{max};
+
+  my $results = [];
+  
+  # search left subtree
+  if ($self->left and $self->left->{max} >= $i->start) {
+    push @{$results}, @{$self->left->search($i)};
+  }
+
+  # search this node
+  push @{$results}, @{$self->_overlapping_intervals($i)};
+
+  # if interval is to the left of the start of this interval, then
+  # it can't be in any child to the right
+  return $results if $i->end < $self->key;
+
+  # search right subtree
+  push @{$results}, @{$self->right->search($i)} if $self->right;
+
+  return $results;
+}
+
+=head2 search_by_key
+
+  Arg [1]     : scalar, $key
+                The key to search the node for
+  Description : Searches for a node by a 'key' value
+  Returntype  : Bio::EnsEMBL::Utils::Tree::Interval::Mutable::Node
+  Exceptions  : none
+  Caller      : general
+
+=cut
+
+sub search_by_key {
+  my ($self, $key) = @_;
+
+  if ($self->key == $key) {
+    return $self;
+  } elsif ($key < $self->key) {
+    return $self->left->search_by_key($key) if $self->left;
+  } else {
+    return $self->right->search_by_key($key) if $self->right;
+  }
+
+}
+
+=head2 insert
+
+  Arg [1]     : Bio::EnsEMBL::Utils::Interval
+                The interval to insert
+  Description : Insert an interval in the node or in its successors
+  Returntype  : none
+  Exceptions  : none
+  Caller      : general
+
+=cut
+
+sub insert {
+  my ($self, $i) = @_;
+
+  if ($i->start < $self->key) {
+    # insert into left subtree
+    unless (defined $self->left) {
+      $self->left(Bio::EnsEMBL::Utils::Tree::Interval::Mutable::Node->new($self->tree, $i));
+      $self->left->parent($self);
+    } else {
+      $self->left->insert($i);
+    }
+  } else {
+    # insert into right subtree
+    unless (defined $self->right) {
```

Comment:
Could this not be swapped so the `if (defined $self->right)` then perform what is in the `else` block otherwise`$self->right(Bio::EnsEMBL::Utils...`